nuntium/models.py

```diff
@@ -303,7 +303,17 @@ def moderate(self):
             raise ValidationError(_('The message needs '
                 + 'to be confirmated first'))
         self.set_to_ready()
-        self.moderation.success()
+        # if we turn on moderation after some messages have been created then
+        # they will not have associated Moderation objects so we need to catch
+        # that, create the moderation object and then try again.
+        try:
+            self.moderation.success()
+        except Moderation.DoesNotExist:
+            if self.writeitinstance.config.moderation_needed_in_all_messages:
+                moderation, created = Moderation.objects.get_or_create(message=self)
```

Comment:
We know the moderation object doesn't exist, so this should just be `Moderation.objects.create()` so people don't wonder about how it might actually exist :)